File: frontend/src/types/csv.ts
export interface CsvColumnSummary {
  name: string;
  detected_type: 'numeric' | 'text' | 'date';
  missing_count: number;
  missing_percentage: number;
  unique_count: number;
  sample_values: string[];
  numeric_min?: number | null;
  numeric_max?: number | null;
  numeric_mean?: number | null;
  numeric_std?: number | null;
}

export interface CsvQualityOverview {
  total_rows: number;
  total_columns: number;
  total_missing_values: number;
  missing_rate_percentage: number;
  duplicate_rows_count: number;
  duplicate_rate_percentage: number;
  numeric_columns_count: number;
  text_columns_count: number;
  date_columns_count: number;
  quality_score: number;
}

export interface CsvAnomaly {
  type: string;
  column: string;
  severity: 'low' | 'medium' | 'high';
  description: string;
  affected_rows_count: number;
}

export interface CsvAiFinding {
  category: string;
  title: string;
  insight: string;
  evidence_columns: string[];
  confidence: number;
}

export interface CsvCompanyDetection {
  detected: boolean;
  company_column?: string | null;
  sample_company_names: string[];
}

export interface CsvAnalysisResponse {
  filename: string;
  file_size_bytes: number;
  quality_overview: CsvQualityOverview;
  columns: CsvColumnSummary[];
  anomalies: CsvAnomaly[];
  ai_findings: CsvAiFinding[];
  company_detection: CsvCompanyDetection;
  parsed_sample_rows: Record<string, unknown>[];
}

File: backend/app/schemas/csv.py
from typing import Any, Dict, List, Optional
from pydantic import BaseModel, Field


class CsvColumnSummary(BaseModel):
    name: str
    detected_type: str
    missing_count: int
    missing_percentage: float
    unique_count: int
    sample_values: List[Any]
    numeric_min: Optional[float] = None
    numeric_max: Optional[float] = None
    numeric_mean: Optional[float] = None
    numeric_std: Optional[float] = None


class CsvQualityOverview(BaseModel):
    total_rows: int
    total_columns: int
    total_missing_values: int
    missing_rate_percentage: float
    duplicate_rows_count: int
    duplicate_rate_percentage: float
    numeric_columns_count: int
    text_columns_count: int
    date_columns_count: int
    quality_score: float  # 0 to 100


class CsvAnomaly(BaseModel):
    type: str
    column: str
    severity: str  # low, medium, high
    description: str
    affected_rows_count: int


class CsvAiFinding(BaseModel):
    category: str
    title: str
    insight: str
    evidence_columns: List[str]
    confidence: float


class CsvCompanyDetection(BaseModel):
    detected: bool
    company_column: Optional[str] = None
    sample_company_names: List[str] = Field(default_factory=list)


class CsvAnalysisResponse(BaseModel):
    filename: str
    file_size_bytes: int
    quality_overview: CsvQualityOverview
    columns: List[CsvColumnSummary]
    anomalies: List[CsvAnomaly]
    ai_findings: List[CsvAiFinding]
    company_detection: CsvCompanyDetection
    parsed_sample_rows: List[Dict[str, Any]]
